tests/h/search/query_test.py

```diff
@@ -97,12 +99,39 @@ def search(self, search):
 
 
 class TestGroupAuthFilter(object):
+    def test_unauthorized_user_can_not_see_the_groups_annotations(self,
+            search, Annotation, group_service):
+        group_service.groupids_readable_by.return_value = []
+        Annotation(groupid="group2").id
+        Annotation(groupid="group1").id
+        Annotation(groupid="group1").id
+
+        result = search.run({})
+
+        assert not result.annotation_ids
+
+    def test_logged_in_group_creator_can_see_the_groups_annotations(self,
+            search, Annotation, group_service):
+        group_service.groupids_readable_by.return_value = ["group1"]
+        Annotation(groupid="group2", shared=True).id
+        users_private_ids = [Annotation(groupid="group1").id,
+                             Annotation(groupid="group1").id]
+
+        result = search.run({})
+
+        assert sorted(result.annotation_ids) == sorted(users_private_ids)
 
     @pytest.fixture
     def search(self, search, pyramid_request):
         search.append_filter(query.GroupAuthFilter(pyramid_request))
         return search
 
+    @pytest.fixture
+    def group_service(self, pyramid_config):
```

Comment:
This fixture has basically the same definition as the `h.search` package's `group_service` fixture. I would suggest simply re-using that fixture and removing this override. This avoids a potential hazard where a change in the name of the package's fixture could cause this one to be used alongside it instead of overriding it, which would break the tests.

If you did want to override a fixture in future to add more complex behaviour, a way to avoid this hazard is to import the "super" fixture as a dependency of the override and then customize its behavior, as we often do with module-specific `pyramid_config` or `pyramid_request` overrides.